datafusion/functions/src/math/log.rs

```diff
@@ -82,7 +82,13 @@ impl ScalarUDFImpl for LogFunc {
     }
 
     fn output_ordering(&self, input: &[ExprProperties]) -> Result<SortProperties> {
-        match (input[0].sort_properties, input[1].sort_properties) {
+        let (base_sort_properties, num_sort_properties) = if input.len() == 1 {
+            // log(x) defaults to log(10, x)
+            (SortProperties::Singleton, input[0].sort_properties)
+        } else {
+            (input[0].sort_properties, input[1].sort_properties)
+        };
+        match (num_sort_properties, base_sort_properties) {
```

Comment:
Looks like the following logic is not completely correct 🤔 
e.g.
base is `DESC NULLS FIRST`
num is `ASC NULLS LAST`
This implementation will return `log(base, num)` is `ASC NULLS LAST`
But the actual output might look like
```
NULL
NULL
1
2
NULL
```
Which is unordered.

Reply:
I agree with you. I noticed that other logic also does not consider `nulls_first`, for example https://github.com/apache/datafusion/blob/36660fe10d9c0cdff62e0da0b94bee28422d3419/datafusion/expr/src/sort_properties.rs#L53

Maybe @berkaysynnada  can help confirm it.

Reply:
You are correct. We also need to check the placement of nulls in `output_ordering()` (and possibly in other implementations of it for multi-input functions). As @jonahgao noticed, `impl SortProperties {}` needs similar handling as well, but that can be addressed in a separate PR. I can create a ticket for it.

Reply:
Fixed the `nulls_first` problem for `log` by https://github.com/apache/datafusion/pull/11571/commits/3feeaa85718e0ce257cceb06004f56ec6267e424. We can create separate PRs for other fixes. 

cc @2010YOUY01  @berkaysynnada